trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -112,41 +114,356 @@ def spin(self,conditions):
+        a_secl = self.airfoil_section_location        
+        rho    = conditions.freestream.density[:,0,None]
+        mu     = conditions.freestream.dynamic_viscosity[:,0,None]
+        Vv     = conditions.frames.inertial.velocity_vector
+        Vh     = self.induced_hover_velocity 
+        a      = conditions.freestream.speed_of_sound[:,0,None]
+        T      = conditions.freestream.temperature[:,0,None]
+        theta  = self.thrust_angle
+        tc     = .12 # Thickness to chord
         
+        BB     = B*B
+        BBB    = BB*B
+            
+        # Velocity in the Body frame
+        T_body2inertial = conditions.frames.body.transform_to_inertial
+        T_inertial2body = orientation_transpose(T_body2inertial)
+        V_body = orientation_product(T_inertial2body,Vv)
         
+        # Velocity transformed to the propulsor frame with flag for tilt rotor
+        if np.isscalar(theta):
+            body2thrust   = np.array([[np.cos(theta), 0., np.sin(theta)],[0., 1., 0.], [-np.sin(theta), 0., np.cos(theta)]])
+            T_body2thrust = orientation_transpose(np.ones_like(T_body2inertial[:])*body2thrust)
+        else:
+            body2thrust = np.zeros((len(theta),3,3))
+            for i in range(len(theta)):
+                body2thrust[i,:,:] = [[np.cos(theta[i][0]), 0., np.sin(theta[i][0])],[0., 1., 0.], [-np.sin(theta[i][0]), 0., np.cos(theta[i][0])]]
+            T_body2thrust      = orientation_transpose(body2thrust)
+       
+        V_thrust      = orientation_product(T_body2thrust,V_body)
         
-        omega1 = self.inputs.omega
+        # Now just use the aligned velocity
+        V = V_thrust[:,0,None]
+        
+        ua = np.zeros_like(V)
+        if Vh != None:   
+            for i in range(len(V)):
+                V_inf = V_thrust[i] 
+                V_Vh =  V_thrust[i][0]/Vh
+                if Vv[i,:].all()  == True :
+                    ua[i] = Vh
+                elif Vv[i][0]  == 0 and  Vv[i][2] != 0: # vertical / axial flight
+                    if V_Vh > 0: # climbing 
+                        ua[i] = Vh*(-(-V_inf[0]/(2*Vh)) + np.sqrt((-V_inf[0]/(2*Vh))**2 + 1))
+                    elif -2 <= V_Vh and V_Vh <= 0:  # slow descent                 
+                        ua[i] = Vh*(1.15 -1.125*(V_Vh) - 1.372*(V_Vh)**2 - 1.718*(V_Vh)**2 - 0.655*(V_Vh)**4 ) 
+                    else: # windmilling 
+                        print("rotor is in the windmill break state!")
+                        ua[i] = Vh*(-(-V_inf[0]/(2*Vh)) - np.sqrt((-V_inf[0]/(2*Vh))**2 + 1))
+                else: # forward flight conditions                 
+                    func = lambda vi: vi - (Vh**2)/(np.sqrt(((-V_inf[2])**2 + (V_inf[0] + vi)**2)))
+                    vi_initial_guess = V_inf[0]
+                    ua[i]    = fsolve(func,vi_initial_guess)
+        else: 
+            ua = 0.0 
+ 
+        ut = 0.0
+        
+        nu    = mu/rho
+        tol   = 1e-5 # Convergence tolerance
```

Comment:
Another hardcoded value.

Reply:
This was always hardcoded. I did not change the tolerance

Reply:
I have not. If I do, this it will be in the feature-BEMT branch were we make a lot of changes to the propeller and rotor models

Reply:
Will do in the BEMT Branch